### DISCUSSION TITLE
New installation, SMTP inbound connection causes crash

### DISCUSSION THREAD

**Discussion by Ramon-LibertyPumps** [postalserver/postal#1946] [Help with using Postal]
[postal_smtp_crash.txt](https://github.com/postalserver/postal/files/8445897/postal_smtp_crash.txt)

I can send email out to an external address using the portal, but when I try to test inbound connections using ([this tool](https://mxtoolbox.com/SuperTool.aspx?action=smtp), I get the above crash in the log after the test times out (about 15 seconds).

This happens whether or not TLS is enabled.

I'm not sure what to go from here.

**Comment by Ramon-LibertyPumps** [comment]
Some additional info: using a different test tool I discovered that if the connecting server is willing to wait more than 25-ish seconds before giving up, then the whole transaction completes normally.

Now to figure out why there is such a long pause after receiving the EHLO...

**Comment by willpower232** [comment]
as Postal has to implement the SMTP protocol in code, there are some situations and circumstances which aren't currently handled gracefully.

I believe this exception can be ignored, it would be nice if the code was able to deal with it like this though
https://github.com/SynergiTech/postal/blob/6f9c7210ff453a6acdaf1bf60f5bcf1e731a7a5b/lib/postal/smtp_server/server.rb#L176

**Reply by Ramon-LibertyPumps** [reply]
Should I ignore it if it's taking so long for my server to respond to the EHLO? I'm not even sure where to begin trying to find the cause, but I don't think it would be wise to go live with this behavior.

**Reply by willpower232** [reply]
You can see in https://github.com/postalserver/postal/blob/main/lib/postal/smtp_server/client.rb that the responses for ehlo and helo don't seem to be complicated so I imagine that if your instance is having a bad time replying promptly, it might be worth seeing how busy the CPU is in case your server is already really busy and can't keep up.
